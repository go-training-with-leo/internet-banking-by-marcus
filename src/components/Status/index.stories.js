import React from 'react';

import Status from '.';

export default {
  title: 'Status',
  component: Status,
};

const Failed = () => <Status failed />;
const Success = () => <Status success />;
const Refund = () => <Status refund />;
const Paid = () => <Status paid />;
const UnPaid = () => <Status unpaid />;
const Canceled = () => <Status canceled />;

export { Canceled, Failed, Paid, Refund, Success, UnPaid };
