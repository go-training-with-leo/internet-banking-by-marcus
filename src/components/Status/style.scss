@import "assets/scss/mixins.scss";
@import "assets/scss/variables.scss";
@import "assets/scss/extends.scss";

$status-colors: (
  "unpaid": #ffd351,
  "canceled": $red-orange,
  "paid": #27ae60,
  "refund": #ffd351,
  "success": #27ae60,
  "failed": $red-orange,
);

.status {
  @include setFlex(row, null, center, 15px, null);
  font-weight: 600;
  font-size: 15px;
  @each $name, $color in $status-colors {
    &.#{$name} {
      color: $color;
    }
  }
}
