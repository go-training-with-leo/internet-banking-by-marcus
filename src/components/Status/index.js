import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

import './style.scss';
import { Cancel, Check, Clock, Reverse } from 'assets/images';

const Status = ({ failed, success, refund, paid, unpaid, cancelled }) => {
  return (
    <div
      className={classNames('status', {
        failed: failed,
        success: success,
        refund: refund,
        paid: paid,
        unpaid: unpaid,
        cancelled: cancelled,
      })}
    >
      {failed ? (
        <>
          <Cancel width={20} height={20} fill='red' />
          <span>Failed</span>
        </>
      ) : success ? (
        <>
          <Check width={20} height={20} fill='green' />
          <span>Success</span>
        </>
      ) : refund ? (
        <>
          <Reverse width={20} height={20} fill='yellow' />
          <span>Refund</span>
        </>
      ) : paid ? (
        <>
          <Check width={20} height={20} fill='green' />
          <span>Paid</span>
        </>
      ) : unpaid ? (
        <>
          <Clock width={20} height={20} fill='yellow' />
          <span>Unpaid</span>
        </>
      ) : (
        cancelled && (
          <>
            <Cancel width={20} height={20} fill='red' />
            <span>Canceled</span>
          </>
        )
      )}
    </div>
  );
};

Status.defaultProps = {
  failed: false,
  success: false,
  refund: false,
  paid: false,
  unpaid: true,
  cancelled: false,
};

Status.propTypes = {
  failed: PropTypes.bool,
  success: PropTypes.bool,
  refund: PropTypes.bool,
  paid: PropTypes.bool,
  unpaid: PropTypes.bool,
  cancelled: PropTypes.bool,
};
export default Status;
