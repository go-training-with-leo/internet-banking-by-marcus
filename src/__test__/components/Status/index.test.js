import React from 'react';
import { render } from '@testing-library/react';
import Status from 'components/Status';

test('Test props children', () => {
  const tree = render(<Status failed />);

  const status = tree.container.querySelector('div.status');
  expect(status).toHaveClass('failed');
  expect(tree).toMatchSnapshot();
});

test('Test pending status', () => {
  const tree = render(<Status refund />);

  const status = tree.container.querySelector('div.status');
  expect(status).toHaveClass('refund');
  expect(tree).toMatchSnapshot();
});

test('Test pending status', () => {
  const tree = render(<Status pending />);

  const status = tree.container.querySelector('div.status');
  expect(status).toHaveClass('pending');
  expect(tree).toMatchSnapshot();
});
