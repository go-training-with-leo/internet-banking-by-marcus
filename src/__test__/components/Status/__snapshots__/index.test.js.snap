// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test pending status 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="status refund unpaid"
      >
        <svg
          fill="#ffd351"
          height="20"
          width="20"
        >
          reverse.svg
        </svg>
        <span>
          Refund
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="status refund unpaid"
    >
      <svg
        fill="#ffd351"
        height="20"
        width="20"
      >
        reverse.svg
      </svg>
      <span>
        Refund
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test pending status 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="status pending unpaid"
      >
        <svg
          fill="#ffd351"
          height="20"
          width="20"
        >
          time.svg
        </svg>
        <span>
          Pending
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="status pending unpaid"
    >
      <svg
        fill="#ffd351"
        height="20"
        width="20"
      >
        time.svg
      </svg>
      <span>
        Pending
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test props children 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="status failed unpaid"
      >
        <svg
          fill="red"
          height="20"
          width="20"
        >
          cancel.svg
        </svg>
        <span>
          Failed
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="status failed unpaid"
    >
      <svg
        fill="red"
        height="20"
        width="20"
      >
        cancel.svg
      </svg>
      <span>
        Failed
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
